### PR TITLE
Implement actions and file flattening

### PR DESCRIPTION
**Pull request by arnisoph**
PR requires #1 and #2 to be merged

### CODE REVIEW

**Comment by arnisoph** [comment]
@hikhvar I've created two issues for the remaining tasks. Since this PR is rather *big* I'd like to the other stuff in dedicated PRs.

File: cmd/trovilo/main.go [hunk 1]
```diff
@@ -94,6 +95,21 @@ func main() {
 						logEntry.Info("Successfully deleted ConfigMap from namepace")
 					}
 
+					// Deleting the configmap also triggers post-deploy actions
+					if len(job.PostDeploy) > 0 {
+						for _, postDeployAction := range job.PostDeploy {
```
Comment (hikhvar) [inline cmd/trovilo/main.go]: We should move this  `for` loop into a dedicated function. It blows up the length of the overall loop. Furthermore we have a similar action in line 162

File: configmap/configmap.go [hunk 2]
```diff
@@ -69,13 +50,13 @@ func VerifyCM(configMap *corev1.ConfigMap, verifySteps []config.VerifyStep) (map
...
 				}
-				err = writeOSFile(tempFile, []byte(fileContents))
+				err = filesystem.WriteOSFile(tempFile, []byte(fileContents))
 				if err != nil {
 					return verifiedFiles, "", err
 				}
 
 				// In the end just remove the temporary file, regardless of the verification result
-				defer deleteFile(tempFile.Name())
+				defer filesystem.DeleteFile(tempFile.Name())
```
Comment (hikhvar) [inline configmap/configmap.go]: This defer should be right beneath the creation of the temporary file. This defer will miss a return in line 55.

**Review by hikhvar — CHANGES_REQUESTED** [review]
Found two improvements. At least the defer should be fixed since we may not delete temporary files in case of an error.

File: configmap/configmap.go (hunk 2 above)
Comment (arnisoph) [reply configmap/configmap.go]: fixed

File: cmd/trovilo/main.go (hunk 1 above)
Comment (arnisoph) [reply cmd/trovilo/main.go]: fixed! thx!

**Review by hikhvar — APPROVED** [review]

**Comment by arnisoph** [comment]
wohooo 🎉

**Pull request merged**